manifests/repo/apt/cloudsmith.pp

```diff
@@ -0,0 +1,49 @@
+# @summary Manages the Erlang repository hosted on cloudsmith for debian based systems.
+#
+# @param [Stdlib::HTTPSUrl] location
+#   Repository location / url
+#
+# @param [String] release
+#   Specifies a distribution of the Apt repository.
+#
+# @param [String] repos
+#   Specifies a component of the Apt repository.
+#
+# @param [String] key
+#   Repository public key id of signage key.
+#
+# @param [Stdlib::HTTPSUrl] key_source
+#   Location / url / source of the repository public key part for signage.
+#
+class erlang::repo::apt::cloudsmith (
+  Stdlib::HTTPSUrl $location,
```

Comment:
instead of using the hiera data, I suggest this:
```suggestion
  Stdlib::HTTPSUrl $location = "https://dl.cloudsmith.io/public/rabbitmq/rabbitmq-erlang/deb/${facts['os']['name']}",
```

that enabled puppet-strings to use the data

Reply:
Is it possible to apply modifiers? In this case i would need it to be downcase or is `${facts['os']['name']}` always supposed to be downcase?

Reply:
I think it is sometimes capotalized, you can wrap it in `downcase()`